File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/evaluation/VisibilityEvaluator.java
package se.gustavkarlsson.aurora_notifier.android.evaluation;

import javax.inject.Inject;

import se.gustavkarlsson.aurora_notifier.android.models.factors.Visibility;

public class VisibilityEvaluator implements ChanceEvaluator<Visibility> {

	@Inject
	public VisibilityEvaluator() {
	}

	public Chance evaluate(Visibility visibility) {
		Integer clouds = visibility.getCloudPercentage();
		if (clouds == null) {
			return Chance.unknown();
		}
		return Chance.of((-1.0 / 50.0) * ((double) clouds) + 1.0);
	}
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/services_impl/evaluation/VisibilityEvaluator.kt
package se.gustavkarlsson.skylight.android.services_impl.evaluation

import se.gustavkarlsson.skylight.android.entities.Chance
import se.gustavkarlsson.skylight.android.entities.Chance.Companion.UNKNOWN
import se.gustavkarlsson.skylight.android.entities.Visibility
import se.gustavkarlsson.skylight.android.services.ChanceEvaluator

class VisibilityEvaluator : ChanceEvaluator<Visibility> {

    override fun evaluate(value: Visibility): Chance {
        val clouds = value.cloudPercentage ?: return UNKNOWN
		if (clouds < 0 || clouds > 100) {
			return UNKNOWN
        }
		val chance = -1.0 / 50.0 * clouds.toDouble() + 1.0
		return Chance(chance)
    }
}
